### PR TITLE
Fix issue for bug #452. Be able to delete / hit backspace on highligh…

### PR DESCRIPTION
**Pull request by jologz**
…t when format attribute is enabled

## Describe the issue/change
There is an issue stated on #452 when a user highlights a group of values that has a format character that it won't delete those grouped values.  

## Describe the changes proposed/implemented in this PR
The change is to let user be able to delete these grouped values but retaining the format character.

## Link Github issue if this PR solved an existing issue
#452 

## Please check which browsers were used for testing

- [X ] Chrome
- [] Safari (OSX)
- [ ] Safari (iOS)
- [ ] Firefox

### CODE REVIEW

File: src/number_format.js [hunk 1]
```diff
@@ -644,11 +637,38 @@ class NumberFormat extends React.Component {
...
+
+      Object.keys(recordIndexOfFormatCharacters).forEach(idx => {
+        if(resolvedPortion.length > idx){
+          resolvedPortion.splice(idx, 0, recordIndexOfFormatCharacters[idx]);
+        } else {
+          resolvedPortion.push(recordIndexOfFormatCharacters[idx])
+        }
+      })
+
+      value = lastValue.substr(0, start) + resolvedPortion.join('');
```
Comment (s-yadav) [inline src/number_format.js:666]: The problem is the format character can have numeric value in itself.
This logic makes the assumption that there will be no numeric char on format itself.

Comment (jologz) [reply src/number_format.js]: My comment was misleading.  I wasn't checking if the characters that were removed were numeric.  I corrected it and also create unit test to back it up

File: src/number_format.js [hunk 2]
```diff
@@ -644,11 +637,37 @@ class NumberFormat extends React.Component {
...
 
-    //if format got deleted reset the value to last value
-    if (this.checkIfFormatGotDeleted(start, end, lastValue)) {
-      value = lastValue;
+    // check whether the deleted portion has a character that is part of a format
+    const deletedValues = lastValue.substr(start, end - start);
+    const hasNonNumeric = !![...deletedValues].find((deletedVal, idx) => this.isCharacterAFormat(idx + start, lastValue));
+
+    // if it has, only remove characters that are not part of the format
+    if(hasNonNumeric) {
```
Comment (s-yadav) [inline src/number_format.js:645]: hasNonNumeric variable name is confusing as format char can have number as well. Can we rename it to formatGotDeleted.

File: test/library/input.spec.js [hunk 3]
```diff
@@ -204,21 +204,28 @@ describe('NumberFormat as input', () => {
...
 
-    //when format and number character are delted
-    wrapper.setProps({value: '+1 (999) 999 9 99 US'})
+    //when a format character is replaced with number
+    wrapper.setProps({value: '+1 (777) 777 7 77 US'});
+    simulateKeyInput(wrapper.find('input'), '8', 8, 9);
+    expect(wrapper.state().value).toEqual('+1 (777) 777 7 77 US');
+  });
+
+  it('should update value if group of characters got deleted with format', () => {
```
Comment (s-yadav) [inline test/library/input.spec.js:213]: When you pull from main repo. You will find a test, `should not update value if formatting got deleted'` which we can remove now.

File: test/library/input.spec.js [hunk 4]
```diff
@@ -204,21 +204,28 @@ describe('NumberFormat as input', () => {
...
-    expect(wrapper.state().value).toEqual('+1 (888) 888 8 88 US');
+    expect(wrapper.state().value).toEqual('+1 (888) 888 8 8  US');
+  })
 
-    //when a format character is replaced with number
-    wrapper.setProps({value: '+1 (777) 777 7 77 US'});
-    simulateKeyInput(wrapper.find('input'), '8', 8, 9);
-    expect(wrapper.state().value).toEqual('+1 (777) 777 7 77 US');
-  });
+  it('should maintain the format even when the format is numeric', () => {
```
Comment (s-yadav) [inline test/library/input.spec.js:224]:
```suggestion
  it('should maintain the format even when the format is numeric and characters are deleted', () => {
```

**Comment by s-yadav** [comment]
Something wrong with your prettier. Can you pull code from main repo. There are some conflicts.

**Comment by jologz** [comment]
My master was outdated, I had to fetch latest and fix merge issues

**Review by s-yadav — APPROVED** [review]

**Comment by s-yadav** [comment]
Can you squash your commits. If I do squash and merge the commit history will get messed up.

**Pull request merged**